### PR TITLE
Code Quality: Clean up DetailsLayoutBrowser related code

### PR DESCRIPTION
**Pull request by 0x5bfa**
## Code Quality: Clean up DetailsLayoutBrowser related code

Refactored before I work on columns refactoring, so I and you can understand quickly and easily how various content views do work.

If you see terrible diff, you can take a look at following classes to see what's going on.

- https://github.com/0x5bfa/Files/blob/5bfa/functionize-columnsview/src/Files.App/Views/LayoutModes/BaseLayout.cs
- https://github.com/0x5bfa/Files/tree/5bfa/functionize-columnsview/src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs

### Motivation & Context

- Improved readability

> **Warning**<br/>Please do not review code lines that I did not changed. I cannot make more diff. If there're more codes that are preferred to change, I will do that and mark that PR as the final one for that stage.

### PR Checklist

- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Related #4180
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?

### Screenshots

_None_

### CODE REVIEW

File: src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs [hunk 1]
```diff
@@ -844,17 +868,18 @@ private void UpdateCheckboxVisibility(object sender, bool isPointerOver)
...
+				if (_userSettingsService.FoldersSettingsService.ShowCheckboxesWhenSelectingItems && control.IsSelected
 					|| isPointerOver)
 					VisualStateManager.GoToState(userControl, "ShowCheckbox", true);
 				else
 					VisualStateManager.GoToState(userControl, "HideCheckbox", true);
 			}
 		}
 
-		// Workaround for https://github.com/microsoft/microsoft-ui-xaml/issues/170
+		#region TextBlock
```
Comment (yaira2) [inline src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs:879]: Please don't use regions

File: src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs [hunk 2]
```diff
@@ -844,17 +868,18 @@ private void UpdateCheckboxVisibility(object sender, bool isPointerOver)
...
 		private void TextBlock_IsTextTrimmedChanged(TextBlock sender, IsTextTrimmedChangedEventArgs e)
 		{
+			// Workaround for https://github.com/microsoft/microsoft-ui-xaml/issues/170
```
Comment (yaira2) [inline src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs:882]: Thank you for adding this 👍

File: src/Files.App/Views/LayoutModes/BaseLayout.cs [hunk 3]
```diff
@@ -1205,11 +1267,6 @@ protected void UninitializeDrag(UIElement element)
 			element.Drop -= Item_Drop;
 		}
 
-		// VirtualKey doesn't support or accept plus and minus by default.
-		public readonly VirtualKey PlusKey = (VirtualKey)187;
-
-		public readonly VirtualKey MinusKey = (VirtualKey)189;
```
Comment (yaira2) [inline src/Files.App/Views/LayoutModes/BaseLayout.cs:1211]: Are these still being used?

**Comment by 0x5bfa** [comment]
If you think diff is messing up, please take a look at https://github.com/0x5bfa/Files/tree/5bfa/functionize-columnsview/src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs to see what is going on.

File: src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs (hunk 1 above)
Comment (0x5bfa) [reply src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs]: Removed

**Comment by lukeblevins** [comment]
@0x5bfa How will this fit with ongoing work in #12601?

**Comment by 0x5bfa** [comment]
OMG. @lukeblevins I wanna work on #12505 asap. When do you expect will #12601 finish?

**Review by ferrariofilippo — COMMENTED** [review]
There are three little things style-related, otherwise LGTM

File: src/Files.App/Views/LayoutModes/BaseLayout.cs [hunk 4]
```diff
@@ -29,124 +29,148 @@
...
 	/// <summary>
-	/// Represents the base class which every layout page must derive from
+	/// Represents the base class which every layout page must derive from.
 	/// </summary>
 	public abstract class BaseLayout : Page, IBaseLayout, INotifyPropertyChanged
 	{
+		protected readonly IUserSettingsService _userSettingsService;
+
+		protected readonly IFileTagsSettingsService _fileTagsSettingsService;
+
```
[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.App/Views/LayoutModes/BaseLayout.cs:39]: I thought it should be applied too. I’ll be fix.

File: src/Files.App/Views/LayoutModes/BaseLayout.cs [hunk 5]
```diff
@@ -159,15 +187,17 @@ public string JumpString
...
 						jumpedToItem = candidateItems.FirstOrDefault();
 					}
 
 					if (jumpedToItem is null)
 					{
 						// Use FilesAndFolders because only displayed entries should be jumped to
 						IEnumerable<ListedItem> candidateItems = ParentShellPageInstance!.FilesystemViewModel.FilesAndFolders
-							.Where(f => f.Name.Length >= value.Length && string.Equals(f.Name.Substring(0, value.Length), value, StringComparison.OrdinalIgnoreCase));
+							.Where(f => f.Name.Length >= value.Length && string.Equals(f.Name[..value.Length], value, StringComparison.OrdinalIgnoreCase));
```
Comment (QuaintMako) [inline src/Files.App/Views/LayoutModes/BaseLayout.cs:199]: The test is a repetition of l.190. Worth creating a named method to simplify the reading?

File: src/Files.App/Views/LayoutModes/BaseLayout.cs [hunk 6]
```diff
@@ -901,7 +961,10 @@ protected void FileList_DragItemsStarting(object sender, DragItemsStartingEventA
 			try
 			{
 				var shellItemList = e.Items.OfType<ListedItem>().Select(x => new VanaraWindowsShell.ShellItem(x.ItemPath)).ToArray();
-				if (shellItemList[0].FileSystemPath is not null && !InstanceViewModel.IsPageTypeSearchResults)
+
+				if (shellItemList[0].FileSystemPath is not null &&
+					InstanceViewModel is not null && 
+					!InstanceViewModel.IsPageTypeSearchResults)
```
Comment (QuaintMako) [inline src/Files.App/Views/LayoutModes/BaseLayout.cs:967]: `!` is redundant with the `null` test the line above?

**Comment by QuaintMako** [comment]
Nice cleaning so far!

File: src/Files.App/Views/LayoutModes/BaseLayout.cs (hunk 5 above)
Comment (0x5bfa) [reply src/Files.App/Views/LayoutModes/BaseLayout.cs:199]: Hmm. I don't think so cause here's already a lot of methods.

File: src/Files.App/Views/LayoutModes/BaseLayout.cs (hunk 6 above)
Comment (0x5bfa) [reply src/Files.App/Views/LayoutModes/BaseLayout.cs:967]: You're recommending `!InstanceViewModel is null`? I don't prefer to use `!`, `==`, and `!=` for null check.

File: src/Files.App/Views/LayoutModes/BaseLayout.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.App/Views/LayoutModes/BaseLayout.cs]: Fixed

**Comment by 0x5bfa** [comment]
Ready for final review.

**Review by ferrariofilippo — APPROVED** [review]
LGTM!

**Review by d2dyno1 — APPROVED** [review]

**Comment by yaira2** [comment]
@0x5bfa can you resolve the merge conflicts?

**Comment by 0x5bfa** [comment]
A moment

**Comment by 0x5bfa** [comment]
I dont want keep this PR because I'm forgetting what I changed and I cannot guarantee my refactoring PR quality. Please let me close this (I'm not going to do here again tho).

**Pull request closed without merging**